cli/cmd/check.go

```diff
@@ -266,39 +269,45 @@ func runExtensionChecks(cmd *cobra.Command, wout io.Writer, werr io.Writer, opts
 			mcCheckCmd := mcCmd.NewCmdCheck()
 			err = executeExtensionCheck(cmd, "multicluster", mcCheckCmd.Flags())
 		default:
-			// Since we don't have checks for extensions we don't support
-			// create a healthchecker that checks if the namespace exists
-			categoryID := healthcheck.CategoryID(ns.Labels[k8s.LinkerdExtensionLabel])
-			checkers := []healthcheck.Checker{}
-			checkers = append(checkers,
-				*healthcheck.NewChecker(fmt.Sprintf("%s extension Namespace exists", categoryID)).
-					WithHintAnchor(fmt.Sprintf("%s-ns-exists", categoryID)).
-					Fatal().
-					WithCheck(func(ctx context.Context) error {
-						_, err := kubeAPI.GetNamespaceWithExtensionLabel(ctx, string(categoryID))
-						if err != nil {
-							return err
-						}
-						return nil
-					}))
-			hc := healthcheck.NewHealthChecker([]healthcheck.CategoryID{categoryID},
-				&healthcheck.Options{
-					ControlPlaneNamespace: controlPlaneNamespace,
-					CNINamespace:          cniNamespace,
-					DataPlaneNamespace:    opts.namespace,
-					KubeConfig:            kubeconfigPath,
-					KubeContext:           kubeContext,
-					Impersonate:           impersonate,
-					ImpersonateGroup:      impersonateGroup,
-					APIAddr:               apiAddr,
-					VersionOverride:       opts.versionOverride,
-					RetryDeadline:         time.Now().Add(opts.wait),
-					CNIEnabled:            opts.cniEnabled,
-				})
-			hc.AppendCategories(*healthcheck.NewCategory(categoryID, checkers, true))
-			success := healthcheck.RunChecks(wout, werr, hc, opts.output)
-			if !success {
-				os.Exit(1)
+			extensionCmd := fmt.Sprintf("linkerd-%s", extension)
+			var path string
+			path, err = exec.LookPath(extensionCmd)
+			if err != nil {
+				if opts.output == healthcheck.JSONOutput {
+					result := map[string]interface{}{
```

Comment:
I'm wondering if we should move print related work from here into `healthcheck_output.go` since other printing related work for the linkerd core checks are in that file. 🤔

Reply:
yeah, that would certainly make sense, but I'm not sure exactly what the best way to structure this is.  I think ideally we'd make all of the various checkOutput structs public so that we could build the check result here in a more typesafe way.  But that's a bigger refactor than I want to tackle right now.